webapp-client/src/component/CategoryList.js

```diff
@@ -1,30 +1,74 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import ErrorMessage from './ErrorMessage';
 
 export default class CategoryList extends React.Component {
-  constructor(props){
+  constructor(props) {
     super(props);
+    this.state = {
+      hasError: false,
+      categoryList: [],
+    };
```

Comment:
would line 29 and 32 work without these lines? just curious- i've no idea

Reply:
Ah they do not hehe, also these lines are important when categorylist is first rendered (so that no error popup shows up)